fix(FEC-12132): Add documentation (#8)
kaltura/playkit-js-ui-managers: README.md

[![Build Status](https://app.travis-ci.com/kaltura/playkit-js-ui-managers.svg?branch=master)](https://app.travis-ci.com/kaltura/playkit-js-ui-managers)
[![](https://img.shields.io/npm/v/@playkit-js/ui-managers/latest.svg)](https://www.npmjs.com/package/@playkit-js/ui-managers)
[![](https://img.shields.io/npm/v/@playkit-js/ui-managers/canary.svg)](https://www.npmjs.com/package/@playkit-js/ui-managers/v/canary)

# playkit-js-ui-managers

playkit-js-ui-managers is a [kaltura player] plugin that groups several UI services,
in order to simplify and facilitate the customization of the [kaltura player] UI by providing a simple and clean API.

> Each service manages a different area of UI functionality

Reviewer: ```suggestion
Each service manages a different area of UI functionality.
```

Author: fixed

It relies on [kaltura player] core API for managing UI features.

playkit-js-ui-managers works as a plugin of [kaltura player]

playkit-js-ui-managers is written in [ECMAScript6] (`*.js`) and [TypeScript] (`*.ts`) (strongly typed superset of ES6), 
and transpiled in ECMAScript5 using [Babel](https://babeljs.io/) and the [TypeScript compiler].

[Webpack] is used to build the distro bundle and serve the local development environment.

[kaltura player]: https://github.com/kaltura/kaltura-player-js.
[ecmascript6]: https://github.com/ericdouglas/ES6-Learning#articles--tutorials
[typescript]: https://www.typescriptlang.org/
[typescript compiler]: https://www.typescriptlang.org/docs/handbook/compiler-options.html
[webpack]: https://webpack.js.org/

## services

> - ###Side Panels

Reviewer: ```suggestion
- Side Panels
```

Reviewer: a space is missing between the number signs and the text

>   manages and controls the side panel items\

Reviewer: ```suggestion
  Manages and controls the side panel items, including:\
```

Author: fix

  \

>   that's includes:

Reviewer: ```suggestion
```

Author: fixed

>   - When an item is activated in the panel, the other items are automatically deactivated in that panel.

Reviewer: ```suggestion
  - When an item is activated in a panel, the other items in that panel are automatically deactivated.
```

Author: fixed

>   - Automatically deactivates a panel when its counter panel is activated\

Reviewer: ```suggestion
  - When a panel is activated, its counter panel is automatically deactivated.\
```

  \
    Full API documentation can be found [here](https://kaltura.github.io/playkit-js-ui-managers/docs/api/classes/side_panels_manager.SidePanelsManager.html)
  

> - ###Upper bar manager

Reviewer: ```suggestion
- Upper bar manager
```

>   manages and controls the upper bar menu\

Reviewer: ```suggestion
  Manages and controls the upper bar menu, including:\
```

  \

>   that's includes:

Reviewer: ```suggestion
```

>   - sets the order icons are shown.

Reviewer: ```suggestion
  - Setting the order in which icons are shown.
```

>   - Increases and decreases the number of icons that appear depending on the size of the player\

Reviewer: ```suggestion
  - Increasing and decreasing the number of icons that appear depending on the size of the player.\
```

Author: fixed
